Further improve README.md (#221)
JuliaPackaging/Yggdrasil: README.md

JLL packages created by Yggdrasil are hosted under the [JuliaBinaryWrappers](https://github.com/JuliaBinaryWrappers) GitHub organization.

We encourage Julia developers to use JLL packages for their libraries.  Here are a few examples of pull requests switching to JLL package:

* [Cairo.jl](https://github.com/JuliaGraphics/Cairo.jl/pull/293)
* [FFTW.jl](https://github.com/JuliaMath/FFTW.jl/pull/122)
* [GSL.jl](https://github.com/JuliaMath/GSL.jl/pull/104)
* [Gtk.jl](https://github.com/JuliaGraphics/Gtk.jl/pull/447)
* [Rsvg.jl](https://github.com/lobingera/Rsvg.jl/pull/36)

You can read more about the `Artifacts` system and JLL packages and how they are important for reproducibility in this post on Julia's blog "[Pkg + BinaryBuilder -- The Next Generation](https://julialang.org/blog/2019/11/artifacts)".

### BinaryProvider.jl

> We hope we convinced you about why it is important to switch to JLL packages.  However, if you really need to support Julia v1.2 or previous versions, you should probably keep using [BinaryProvider.jl](https://github.com/JuliaPackaging/BinaryProvider.jl) instead.  As explained in the previous section, a `build.jl` file is no more generated after the build as that is not used for JLL packages, instead you would need to generate it yourself.  In the top-directory of Yggdrasil there is a script to do this: [`generate_buildjl.jl`](./generate_buildjl.jl).  It takes three positional arguments:

Reviewer: ```suggestion
We hope we convinced you about why it is important to switch to JLL packages.  However, if you really need to support Julia v1.2 or previous versions, you should keep using [BinaryProvider.jl](https://github.com/JuliaPackaging/BinaryProvider.jl).  As explained in the previous section, a `build.jl` file is not generated after the build anymore as that is not used for JLL packages, instead you would need to generate it yourself: In the top-directory of Yggdrasil there is a script to do this: [`generate_buildjl.jl`](./generate_buildjl.jl).  It takes three positional arguments:
```

* the path to the `build_tarballs.jl` script

> * the `owener/name` of the repository where the tarballs have been uploaded.  If omitted, this defaults to `JuliaBinaryWrappers/LIBRARY-NAME_jll.jl`

Reviewer: ```suggestion
* the `owner/name` of the repository where the tarballs have been uploaded.  If omitted, this defaults to `JuliaBinaryWrappers/BuilderName_jll.jl`
```

> * the tage name where the tarballs have been uploaded.  If omitted, this defaults to the latest version of the JLL package in the [General registry](https://github.com/JuliaRegistries/General).  If there are no versions of the package in the registry, the script will fail.

Reviewer: ```suggestion
* the tag name where the tarballs have been uploaded.  If omitted, this defaults to the latest version of the JLL package in the [General registry](https://github.com/JuliaRegistries/General).  If there are no versions of the package in the registry, the script will fail.
```

For example, to get the `build.jl` file for the latest version of Zlib you can run the following command:

```
julia --color=yes generate_buildjl.jl Z/Zlib/build_tarballs.jl
```

If instead you want to get the `build_tarballs.jl` file for the tag named [Zlib-v1.2.11+6](https://github.com/JuliaBinaryWrappers/Zlib_jll.jl/releases/tag/Zlib-v1.2.11%2B6) you have to run the command

```
julia --color=yes generate_buildjl.jl Z/Zlib/build_tarballs.jl JuliaBinaryWrappers/Zlib_jll.jl Zlib-v1.2.11+6
```

> *Note*: you have to manually add `prefix` as the first argument to all `Product` constructors in the generated `build.jl` files.

Reviewer: ```suggestion
*Note*: you have to manually add `prefix` as the first argument to all `Product` constructors in the generated `build.jl` files.  This is necessary because the syntax between `BinaryBuilder v0.2+` and `BinaryProvider` has diverged.
```

> Remember that you will need the `build.jl` file also for all direct and indirect dependencies of the binary library.

Reviewer: ```suggestion
Remember that you will also need the `build.jl` files for all direct and indirect dependencies.
```

> Here are a few examples of packages using this system to install the libraries:

Reviewer: ```suggestion
Here are a few examples of packages using this system to install their libraries:
```
